src/CheckBox/CheckBox.tsx

```diff
@@ -25,7 +26,7 @@ export const CheckBox: React.FunctionComponent<CheckBoxProps> = (props: CheckBox
                     <input
                         type="checkbox"
                         className="custom-control-input"
-                        id={props.name}
+                        id={props.id || props.name}
```

Comment:
Id should be random if not set. Name and ID should be separated.

Reply:
resolved in commit https://github.com/sebgroup/react-components/pull/31/commits/cd1e9c4a4dbca52693e196eeec05a0303300e026